### PR TITLE
member: return err when meet frequently campaign leader

### LINKED ISSUE #7562: panic: keep lease panic
## Bug Report
CampaignLeader maybe pass if the retry time has reached campaignLeaderFrequencyTimes, but the lease maybe not success ,so the lease time is zero .
https://github.com/tikv/pd/blob/cf718a799fcddda5413e6e8c8df0a14270b3d20e/server/server.go#L1645-L1655

### What did you expect to see?
normal
### What did you see instead?
```
"[log.go:87] [panic] [recover=\"\\\"non-positive interval for NewTicker\\\"\"] [stack=\"[github.com/tikv/pd/pkg/utils/logutil.LogPanic\\n\\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/pd/pkg/utils/logutil/log.go:87\\nruntime.gopanic\\n\\t/usr/local/go/src/runtime/panic.go:914\\ntime.NewTicker\\n\\t/usr/local/go/src/time/tick.go:22\\ngithub.com/tikv/pd/pkg/election.(*lease).keepAliveWorker.func1\\n\\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/pd/pkg/election/lease.go:152](http://github.com/tikv/pd/pkg/utils/logutil.LogPanic//n//t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/pd/pkg/utils/logutil/log.go:87//nruntime.gopanic//n//t/usr/local/go/src/runtime/panic.go:914//ntime.NewTicker//n//t/usr/local/go/src/time/tick.go:22//ngithub.com/tikv/pd/pkg/election.(*lease).keepAliveWorker.func1//n//t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/pd/pkg/election/lease.go:152)\"]"
```
### What version of PD are you using (`pd-server -V`)?
8196d84c0

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #7562

### What is changed and how does it work?

```commit-message
```
CampaignLeader maybe pass if the retry time has reached campaignLeaderFrequencyTimes, but the lease maybe not success ,so the lease time is zero .

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]

File: pkg/member/member.go [hunk 1]
```diff
@@ -185,12 +185,16 @@ func (m *EmbeddedEtcdMember) CampaignLeader(ctx context.Context, leaseTimeout in
...
+
 	if m.leadership.GetCampaignTimesNum() >= campaignLeaderFrequencyTimes {
-		log.Warn("campaign times is too frequent, resign and campaign again",
-			zap.String("leader-name", m.Name()), zap.String("leader-key", m.GetLeaderPath()))
 		m.leadership.ResetCampaignTimes()
-		return m.ResignEtcdLeader(ctx, m.Name(), "")
+		if err := m.ResignEtcdLeader(ctx, m.Name(), ""); err != nil {
+			return err
+		}
+		return errors.Errorf("campaign times is too frequent, resign and campaign again, "+
```
Comment (JmPotato) [inline pkg/member/member.go:194]: Do we need to define it in `pkg/errs/errno.go`?

File: pkg/election/leadership.go [hunk 2]
```diff
@@ -156,6 +157,16 @@ func (ls *Leadership) Campaign(leaseTimeout int64, leaderData string, cmps ...cl
 		lease:   clientv3.NewLease(ls.client),
 	}
 	ls.setLease(newLease)
+
+	failpoint.Inject("skipGrant", func(val failpoint.Value) {
```
Comment (JmPotato) [inline pkg/election/leadership.go:161]: Suggest using name `checkGrantMember`.

Comment (HuSharp) [reply pkg/election/leadership.go:161]: maybe `skipGrantLeader`?

File: pkg/member/member.go (hunk 1 above)
Comment (HuSharp) [reply pkg/member/member.go:194]: good idea, how about `ErrEtcdLeaderFrequentlyChange`

**Review by JmPotato — APPROVED** [review]

**Pull request merged**